jwplatform/client.py

```diff
@@ -219,28 +256,81 @@ def clip(self, site_id, channel_id, event_id, body=None, query_params=None):
         )
 
 
-class _MediaClient(_SiteResourceClient):
+CREATE_MEDIA_PAYLOAD = {
+    "upload": {
+    },
+    "metadata": {
+
+    }
+}
 
+
+class _MediaClient(_SiteResourceClient):
     _resource_name = "media"
     _id_name = "media_id"
 
+    def __init__(self, client: JWPlatformClient):
+        super().__init__(client)
+
     def reupload(self, site_id, body, query_params=None, **kwargs):
         resource_id = kwargs[self._id_name]
         return self._client.request(
             method="PUT",
-            path=self._singular_path.format(site_id=site_id, resource_name=self._resource_name, resource_id=resource_id) + "reupload/",
+            path=self._singular_path.format(site_id=site_id, resource_name=self._resource_name,
+                                            resource_id=resource_id) + "reupload/",
             body=body,
             query_params=query_params
         )
 
-class _WebhookClient(_ResourceClient):
+    def get_upload_handler(self, site_id, file, body=None, query_params=None, **kwargs):
+        if not kwargs:
+            kwargs = {}
+
+        # Determine the upload type - Single or multi-part
+        target_part_size = int(kwargs['target_part_size']) if 'target_part_size' in kwargs else constants.MIN_PART_SIZE
+        upload_method = determine_upload_method(file, target_part_size)
+        if not body:
+            body = CREATE_MEDIA_PAYLOAD.copy()
+        body["upload"]["method"] = upload_method
+
+        # Create the media
+        resp = self.create(site_id, body, query_params)
+
+        # Upload the file
+        upload_handler = self.process_upload(resp, upload_method, file, **kwargs)
+
+        # Return the upload_instance to the caller so that they can resume at their own by calling the upload
+        #  again
+        return upload_handler
+
+    def process_upload(self, resp, upload_method, file, **kwargs):
+        base_url = kwargs['base_url'] if 'base_url' in kwargs else None
+        target_part_size = int(kwargs['target_part_size']) if 'target_part_size' in kwargs else constants.MIN_PART_SIZE
+        retry_count = int(kwargs['retry_count']) if 'retry_count' in kwargs else constants.RETRY_COUNT
+
+        if upload_method == UploadType.direct.value:
+            result = resp.json_body
+            direct_link = result["upload_link"]
+            upload_handler = SingleUpload(direct_link, file, retry_count)
+        elif upload_method == UploadType.multipart.value:
+            result = resp.json_body
+            upload_id = result["upload_id"]
+            upload_token = result["upload_token"]
+            upload_client = _UploadClient(api_secret=upload_token, base_url=base_url) \
+                if base_url else _UploadClient(api_secret=upload_token)
```

Comment:
Might make sense for `_UploadClient`'s constructor to handle the case where this is `None`.